packages/components/src/popover/index.js

```diff
@@ -42,405 +49,140 @@ const SLOT_NAME = 'Popover';
-
-	return offsetIframe(
-		parentNode.getBoundingClientRect(),
-		parentNode.ownerDocument,
-		container
-	);
-}
+	return y;
+};
 
-/**
- * Sets or removes an element attribute.
- *
- * @param {Element} element The element to modify.
- * @param {string}  name    The attribute name to set or remove.
- * @param {?string} value   The value to set. A falsy value will remove the
- *                          attribute.
- */
-function setAttribute( element, name, value ) {
-	if ( ! value ) {
-		if ( element.hasAttribute( name ) ) {
-			element.removeAttribute( name );
+const placementToAnimationOrigin = ( placement ) => {
+	const [ a, b ] = placement.split( '-' );
+
+	let x, y;
+	if ( a === 'top' || a === 'bottom' ) {
+		x = a === 'top' ? 'bottom' : 'top';
+		y = 'middle';
+		if ( b === 'start' ) {
+			y = 'left';
+		} else if ( b === 'end' ) {
+			y = 'right';
 		}
-	} else if ( element.getAttribute( name ) !== value ) {
-		element.setAttribute( name, value );
 	}
-}
 
-/**
- * Sets or removes an element style property.
- *
- * @param {Element} element  The element to modify.
- * @param {string}  property The property to set or remove.
- * @param {?string} value    The value to set. A falsy value will remove the
- *                           property.
- */
-function setStyle( element, property, value = '' ) {
-	if ( element.style[ property ] !== value ) {
-		element.style[ property ] = value;
-	}
-}
-
-/**
- * Sets or removes an element class.
- *
- * @param {Element} element The element to modify.
- * @param {string}  name    The class to set or remove.
- * @param {boolean} toggle  True to set the class, false to remove.
- */
-function setClass( element, name, toggle ) {
-	if ( toggle ) {
-		if ( ! element.classList.contains( name ) ) {
-			element.classList.add( name );
+	if ( a === 'left' || a === 'right' ) {
+		x = 'center';
+		y = a === 'left' ? 'right' : 'left';
+		if ( b === 'start' ) {
+			x = 'top';
+		} else if ( b === 'end' ) {
+			x = 'bottom';
 		}
-	} else if ( element.classList.contains( name ) ) {
-		element.classList.remove( name );
-	}
-}
-
-function getAnchorDocument( anchor ) {
-	if ( ! anchor ) {
-		return;
-	}
-
-	if ( anchor.endContainer ) {
-		return anchor.endContainer.ownerDocument;
 	}
 
-	if ( anchor.top ) {
-		return anchor.top.ownerDocument;
-	}
-
-	return anchor.ownerDocument;
-}
+	return x + ' ' + y;
+};
 
 const Popover = (
 	{
+		// Disable reason: We generate the `...contentProps` rest as remainder
+		// of props which aren't explicitly handled by this component.
+		/* eslint-disable no-unused-vars */
+		range,
+		/* eslint-enable no-unused-vars */
+		animate = true,
 		headerTitle,
 		onClose,
 		children,
 		className,
 		noArrow = true,
 		isAlternate,
-		// Disable reason: We generate the `...contentProps` rest as remainder
-		// of props which aren't explicitly handled by this component.
-		/* eslint-disable no-unused-vars */
-		position = 'bottom right',
-		range,
+		position,
+		placement = 'bottom-start',
```

Comment:
It looks like the allowed values for the `placement` prop are different from the ones used previously (I guess because we swapped from `popper` to `floating-ui`).

This may introduce a breaking change, and therefore I suggest we keep accepting the same values as before, but instead we do an internal conversion from "popper value" to "floating-ui value" (e.g `'bottom-left' => 'left-start'`).

WDYT ?

Reply:
🤦  You're correct, I mixed `placement` and `position`.

So, to recap:

- `position` doesn't have a default value anymore (used to be `bottom-right`)
- `placement` is added as a new prop
- the `positionToPlacement` function "translates" a value expressed thought the `position` prop, to a value for the `placement` prop

Should we look into deprecating the `position` prop?

Reply:
> Should we look into deprecating the position prop?

I considered this but thought it's too soon here. See my comment https://github.com/WordPress/gutenberg/pull/40740#issuecomment-1117123413

> since the default value of position was bottom-right, shouldn't the new default value for placement be bottom-end ?

haha, I'm not sure any more these things confuse me. I think "bottom right" in position is equivalent to "bottom-start". Because it means the popover is at the bottom but the popover opens to the right (or the anchor is at the start of the popover) but I can be wrong, I'll check again.

> we should update the Popover's README too

Good point.

Reply:
> haha, I'm not sure any more these things confuse me. I think "bottom right" in position is equivalent to "bottom-start". Because it means the popover is at the bottom but the popover opens to the right (or the anchor is at the start of the popover) but I can be wrong, I'll check again.

now I'm even more confused 😅 

Looking at `floating-ui`'s [landing page](https://floating-ui.com/), it looks like `bottom-right` should be `bottom end` (at least in a LTR context)

![Screenshot 2022-05-10 at 23 16 00](https://user-images.githubusercontent.com/1083581/167723213-92140651-ac50-41b5-8271-714d6e9b0793.png)

Although, as you point out, in that case the popover is _anchored_ to the bottom right, but "expands" left

Reply:
yeah, I did more test and bottom-start is the right one here (same result as position bottom right)